manga_ocr_dev/synthetic_data_generator_v2/generator.py

```diff
@@ -195,16 +195,18 @@ def process(self, text=None, override_params=None):
         font_path = params.get("font_path")
         if font_path:
             vocab = self.font_map.get(font_path)
-            if vocab:
-                unsupported_chars = {c for c in text_gt if c not in vocab and not c.isspace()}
-                if unsupported_chars:
-                    try:
-                        # Attempt to find a fallback font that supports all characters
-                        params["font_path"] = self.get_random_font(text_gt)
-                    except ValueError:
-                        # If no fallback is found, strip unsupported characters and proceed
-                        text_gt = "".join([c for c in text_gt if c in vocab or c.isspace()])
-                        lines = self.words_to_lines(self.split_into_words(text_gt))
+            if vocab is None:
+                vocab = set()
+
+            unsupported_chars = {c for c in text_gt if c not in vocab and not c.isspace()}
+            if unsupported_chars:
+                try:
+                    # Attempt to find a fallback font that supports all characters
+                    params["font_path"] = self.get_random_font(text_gt)
```

Comment:
instead of randomly picking a font, the user should supply a default fallback font.